### PR TITLE
Add AutoPackets documentation and fix navigation issue hiding Multiplayer Gem docs

### PR DESCRIPTION
**Pull request by AMZN-puvvadar**
This adds documentation around AutoPackets (we have docs for AutoComponents but not Packets) and fixes an issue I observed where the Multiplayer Gem doesn't appear in the left hand navigation bar. The content is searchable but can't be obviously navigated to.

### CODE REVIEW

File: content/docs/user-guide/gems/reference/multiplayer/_index.md [hunk 1]
```diff
@@ -14,10 +14,11 @@ For a quick introduction of the O3DE Multiplayer Gem and networking, watch the v
 
 | Topic | Description |
 |---|---|
-| [Overview](./overview) | An overview of the O3DE Multiplayer Gem, how it works, and how to use it in your project. |
-| [Project configuration](./configuration) | How to enable the O3DE Multiplayer Gem in a project. |
-| [Multiplayer Autocomponents](./autocomponents) | Automatically create components for use with the Multiplayer Gem using the AzAutoGen system. |
-| [Test Multiplayer Projects in the Editor](./test-in-editor) | Automatically launch local servers or connect to a remote persistent server when working on a multiplayer project in the O3DE Editor. |
+| [Overview](./multiplayer-gem/overview) | An overview of the O3DE Multiplayer Gem, how it works, and how to use it in your project. |
+| [Project configuration](./multiplayer-gem/configuration) | How to enable the O3DE Multiplayer Gem in a project. |
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/multiplayer/_index.md:18]: nit: Drop the O3DE part of "O3DE Multiplayer Gem" here as you drop it for all later lines.

File: content/docs/user-guide/gems/reference/multiplayer/_index.md [hunk 2]
```diff
@@ -14,10 +14,11 @@ For a quick introduction of the O3DE Multiplayer Gem and networking, watch the v
...
+| [Multiplayer AutoComponents](./multiplayer-gem/autocomponents) | Automatically create components for use with the Multiplayer Gem using the AzAutoGen system. |
+| [Multiplayer AutoPackets](./multiplayer-gem/autopackets) | Define and automatically create packets for use with the Multiplayer Gem using the AzAutoGen system. |
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/multiplayer/_index.md:20]: Is there a descriptive word to place in front of "packets". From this sentence its unclear why I need to do/should do this.  But what is a "packet" in this context?

File: content/docs/user-guide/gems/reference/multiplayer/_index.md [hunk 3]
```diff
@@ -14,10 +14,11 @@ For a quick introduction of the O3DE Multiplayer Gem and networking, watch the v
 
 | Topic | Description |
 |---|---|
-| [Overview](./overview) | An overview of the O3DE Multiplayer Gem, how it works, and how to use it in your project. |
-| [Project configuration](./configuration) | How to enable the O3DE Multiplayer Gem in a project. |
-| [Multiplayer Autocomponents](./autocomponents) | Automatically create components for use with the Multiplayer Gem using the AzAutoGen system. |
-| [Test Multiplayer Projects in the Editor](./test-in-editor) | Automatically launch local servers or connect to a remote persistent server when working on a multiplayer project in the O3DE Editor. |
+| [Overview](./multiplayer-gem/overview) | An overview of the O3DE Multiplayer Gem, how it works, and how to use it in your project. |
+| [Project configuration](./multiplayer-gem/configuration) | How to enable the O3DE Multiplayer Gem in a project. |
+| [Multiplayer AutoComponents](./multiplayer-gem/autocomponents) | Automatically create components for use with the Multiplayer Gem using the AzAutoGen system. |
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/multiplayer/_index.md:19]: IMHO these are technical links, but they do not provide any context for why a developer wants to do this thing/

What is a "component" here? Firstly, component is standard terminology  from Lumberyard/O3DE days, but this is really an Auto component. Hard to tell from sentence the why here. Maybe these should be less technical and more informative ie "Create  components that automate network synchronization for your networking application"?

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
+---
+title: Multiplayer AutoPackets
+description: A reference for defining Open 3D Engine multiplayer packets through autopackets.
+linktitle: AutoPackets
+---
+
+*AutoPackets* provide a convenient way to define network packets used to communicate data between endpoints in a Multiplayer session. Using the [AzAutoGen](/docs/user-guide/engine/autogen) system, autopacket files found inside of your project are processed during builds to create C++ classes and handlers for packets with defined payloads.
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md:7]: To me its not clear what an auto-packet is here? Are we defining a protocol for communication? Are they are collecting of attributes for a packet?

Could you provide an example(s) of a 'typical' autopacket here ie a Connect packet that contains the properties expected during connection events such as a protocol version and ticket id?

**Review by lmbr-pip — COMMENTED** [review]
LG, mostly.

Feel some of this text jumps straight to technical details without setting enough context for folks new to O3DE networking. I would try and define what an AutoComponent and AutoPacket are and could be used for in a general way, before getting into the technical specifics of O3DE

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autocomponents.md [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
-linktitle: Autocomponents
+linktitle: AutoComponents
 ---
 
-*Autocomponents* provide a convenient way to define states of a multiplayer component relevant to network synchronization. Using the [AzAutoGen](/docs/user-guide/engine/autogen) system, autocomponent files found inside of your project are processed during builds to create C++ classes for components and controllers that provide network replication and remote function calls. Autocomponents also take care of Editor and behavior context bindings so that the bound component shows in the Editor and works with O3DE scripting.
+*AutoComponents* provide a convenient way to define states of a multiplayer component relevant to network synchronization. Using the [AzAutoGen](/docs/user-guide/engine/autogen) system, autocomponent files found inside of your project are processed during builds to create C++ classes for components and controllers that provide network replication and remote function calls. AutoComponents also take care of Editor and behavior context bindings so that the bound component shows in the Editor and works with O3DE scripting.
```
Comment (AMZN-Gene) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autocomponents.md:7]: I think we should spell autocomponent as auto-component.  pinging @sptramer

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
+---
+title: Multiplayer AutoPackets
```
Comment (AMZN-Gene) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md:2]: Similar to auto-components I think AutoPacket = Auto-Packet or Auto-packet?

**Review by AMZN-Gene — APPROVED** [review]

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autocomponents.md (hunk 5 above)
Comment (AMZN-puvvadar) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autocomponents.md:7]: I went with our in code naming but I'm open here. I just felt like Autocomponents and Autopackets didn't necessarily read well.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+### Packet attributes
+
+The `Packet` tag defines the name, description and handshake relevance of the packet
+
+| Property | Description | Type |
+|---|---|---|---|
+| Name | The class name of the generated autopacket. | `string`: Must be a valid C++ class name. |
+| Desc | The description of the generated autopacket. | `string`: A user friendly string description of the packet. |
+| HandshakePacket | If the packet is part of a connection handshake. | `string`: Only packets with HandshakePacket as true can be processed while the user implemented IsHandshakeComplete function returns false. |
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md:32]:
```suggestion
The `Packet` tag defines a new packet.

| Property | Description | Type |
|---|---|---|---|
| Name | The class name of the generated autopacket. | A valid C++ class identifier. |
| Desc | The description of the generated autopacket. | `string` |
| HandshakePacket | If the packet is part of a connection handshake. | `bool` |

{{< note >}}
Packets with `HandshakePacket ` set as true will only be processed if the user-implemented `IsHandshakeComplete` function for the packet returns `false`.
{{< /note >}}
```

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+| Property | Description | Type |
+|---|---|---|---|
+| Name | The name of the packet member. | `string`: Must be a valid C++ variable name. |
+| Type | The type of the packet member. | `string`: Must be a valid C++ type. |
+| Init | The initial value of the packet member. | `<Type>`: Must be a valid value of `<Type>`. |
+| Container | If specified, the member represents a container of `<Type>`. | `Vector`, `Array`|
+| | **Vector**: A resizeable vector. |
+| | **Array**: A fixed size array. | |
+| Count | If Container is specified, the size of the container. | `uint` |
```
Comment (sptramer) [inline content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md:46]:
```suggestion
The `Member` tag defines data on the packet. This is the primary mechanism for defining a packet's payload.

| Property | Description | Type |
|---|---|---|---|
| Name | The name of the packet member. | A valid C++ variable name. |
| Type | The type of the packet member. | A valid C++ type. |
| Init | The initial value of the packet member. | `<Type>` |
| Container | If provided, the member is a container of `<Type>`. | `Vector`, `Array`|
| | **Vector**: A resizeable vector. |
| | **Array**: A fixed size array. | |
| Count | If Container is provided, the size of the container. | `uint` |
```

**Review by sptramer — CHANGES_REQUESTED** [review]
* Breaks the link on `gems/reference/_index.md` to the Multiplayer gem

Some suggestions to update the descriptions on the auto-packets. I think I got the technical details correct based on a quick source browse but might have missed something.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autocomponents.md [hunk 9]
```diff
@@ -1,16 +1,16 @@
...
 
 In order to enable autocomponent builds for your project, follow the instructions in [Multiplayer Project Configuration](./configuration).
 
 ## Autocomponent file structure
 
-Autocomponents are defined in XML files, placed in the `Code\Source\Autogen` directory of the Multiplayer Gem.
+AutoComponents are defined in XML files, placed in the `Code\Source\Autogen` directory of the Multiplayer Gem.
```
[the remark this answers is not in the document]
Comment (AMZN-puvvadar) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autocomponents.md:13]: Ah yeah, they can be anywhere using AzNetworking technically. I should probably restructure this.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+title: Multiplayer AutoPackets
+description: A reference for defining Open 3D Engine multiplayer packets through autopackets.
+linktitle: AutoPackets
+---
+
+*AutoPackets* provide a convenient way to define network packets used to communicate data between endpoints in a Multiplayer session. Using the [AzAutoGen](/docs/user-guide/engine/autogen) system, autopacket files found inside of your project are processed during builds to create C++ classes and handlers for packets with defined payloads.
+
+In order to enable autopacket builds for your project, follow the instructions in [Multiplayer Project Configuration](./configuration).
+
+## AutoPacket file structure
```
[the remark this answers is not in the document]
Comment (AMZN-puvvadar) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md:11]: Not to my knowledge, unless there was one for AutoComponents that's not apparent to me?

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+
+AutoPackets are defined in XML files, placed in the `Code\Source\Autogen` directory of the Multiplayer Gem.
```
[the remark this answers is not in the document]
Comment (AMZN-puvvadar) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md:13]: They can be anywhere making use of AzNetworking including AzNetworking itself, I'll do some restructuring here.

File: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md (hunk 4 above)
Comment (AMZN-puvvadar) [reply content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/autopackets.md:7]: There's an example detailing that further down in the doc.

File: content/docs/user-guide/gems/reference/multiplayer/_index.md (hunk 3 above)
Comment (AMZN-puvvadar) [reply content/docs/user-guide/gems/reference/multiplayer/_index.md:19]: That's really feedback for @sptramer and @willihay . I was just adding a link here though I'll soon be migrating it to AzNetworking.

File: content/docs/user-guide/gems/reference/multiplayer/_index.md (hunk 2 above)
Comment (AMZN-puvvadar) [reply content/docs/user-guide/gems/reference/multiplayer/_index.md:20]: I'm moving this to AzNetworking where it'll make a lot more sense.

**Comment by AMZN-puvvadar** [comment]
Feedback is largely addressed. I handled the naming format for auto-component and auto-packet and moved the auto-packet doc to networking where it should have been in the first place. I also added links and took much of @sptramer 's formatting and content suggestions.

**Review by sptramer — APPROVED** [review]

File: content/docs/user-guide/gems/reference/_index.md [hunk 12]
```diff
@@ -123,7 +123,6 @@ toc: true
 | [Certificate Manager](./network/certificate-manager) | The Certificate Manager Gem provides access to authentication files for secure game connections from Amazon S3, files on disk, and other 3rd party sources. |
 | [Http Requestor](./network/http-requestor) | The HTTP Requestor Gem provides functionality to make asynchronous HTTP/HTTPS requests and return data through a user-provided call back function. |
 | [Metastream](./network/twitch/metastream) | The Metastream Gem provides functionality for an HTTP server that allows broadcasters to customize game streams with overlays of statistics and event data from a game session. |
-| [Multiplayer](./network/multiplayer) | The Multiplayer Gem provides support for synchronizing O3DE components across a network in a fault-tolerant way, allowing for the development of multiplayer games. |
```
Comment (lmbr-pip) [inline content/docs/user-guide/gems/reference/_index.md:126]: Did you mean to remove this reference? Don't we want a reference to the Multiplayer Gem here?

File: content/docs/user-guide/networking/autopackets.md [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+
+In order to enable auto-packet builds for your project, you must add AZ::AzNetworking as a build dependency of your project.
+
+## Auto-packet file structure
+
+Auto-packets are defined in XML files, placed in the `Code\Source\Autogen` directory of the relevant project or Gem.
+
+### PacketGroup attributes
+
+The top-level element of an auto-packet definition is a `PacketGroup`. Packet groups associate different types of individual packet definitions together. Packets defined as part of a packet group are placed into a namespace.
```
Comment (lmbr-pip) [inline content/docs/user-guide/networking/autopackets.md:17]: nit: "the same namespace" or " the packet group's namespace"? to be clearer.

File: content/docs/user-guide/networking/autopackets.md [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+| Name | The namespace for generated auto-packets. | A valid C++ namespace identifier |
+| PacketStart | The value to start sequencing packet type identifiers for this packet group at. To avoid conflicts with the core AzNetworking framework, use `CorePackets::PacketType::Max`. | `const int32` |
+
+### Packet attributes
+
+The `Packet` tag defines a new packet.
+
+| Property | Description | Type |
+|---|---|---|---|
+| Name | The class name of the generated autopacket. | A valid C++ class identifier. |
```
Comment (lmbr-pip) [inline content/docs/user-guide/networking/autopackets.md:30]: Do any of these fields have size/limits on the type?

File: content/docs/user-guide/networking/autopackets.md [hunk 15]
```diff
@@ -0,0 +1,151 @@
...
+Packets with `HandshakePacket ` set as true will only be processed if the user-implemented `IsHandshakeComplete` function for the packet returns `false`.
+{{< /note >}}
+
+### Member attributes
+
+The `Member` tag defines data on the packet. This is the primary mechanism for defining a packet's payload.
+
+| Property | Description | Type |
+|---|---|---|---|
+| Name | The name of the packet member. | A valid C++ variable name. |
```
Comment (lmbr-pip) [inline content/docs/user-guide/networking/autopackets.md:43]: Do any of these fields have size/limits on the type?

File: content/docs/user-guide/networking/autopackets.md [hunk 16]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+### Packet Handling
+
+With the exception of CorePackets defined in AzNetworking, packets are handled via the IConnectionListener interface. Implementers of this interface typically define the following:
+
+```C++
+    //! IConnectionListener interface
+    //! @{
+    AzNetworking::ConnectResult ValidateConnect(const AzNetworking::IpAddress& remoteAddress, const AzNetworking::IPacketHeader& packetHeader, AzNetworking::ISerializer& serializer) override;
```
Comment (lmbr-pip) [inline content/docs/user-guide/networking/autopackets.md:120]: nit: Consider new lines between methods to make this code block more readable

**Review by lmbr-pip — COMMENTED** [review]
LG.

Have some minor comments.

File: content/docs/user-guide/gems/reference/_index.md (hunk 12 above)
Comment (AMZN-puvvadar) [reply content/docs/user-guide/gems/reference/_index.md:126]: If you look up a bit, it's in the Multiplayer section right above Networking. It seemed redundant to have it in both.

File: content/docs/user-guide/networking/autopackets.md (hunk 14 above)
Comment (AMZN-puvvadar) [reply content/docs/user-guide/networking/autopackets.md:30]: Nope, should just be the limitations of the type itself.

File: content/docs/user-guide/networking/autopackets.md (hunk 15 above)
Comment (AMZN-puvvadar) [reply content/docs/user-guide/networking/autopackets.md:43]: Nope, these aren't specialized types so there's nothing explicit. Granted larger counts on containers will yield larger packets.

**Review by lmbr-pip — APPROVED** [review]

**Pull request merged**